server/status.go

```diff
@@ -152,19 +154,98 @@ func (s *statusServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	s.router.ServeHTTP(w, r)
 }
 
-// handleGossipStatus handles GET requests for gossip network status.
-func (s *statusServer) handleGossipStatus(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
-	w.Header().Set(util.ContentTypeHeader, util.JSONContentType)
+// extractNodeID examines the node_id URL parameter and returns the nodeID and a
+// boolean showing if it is this node. If node_id is "local" or not present, it
+// returns the local nodeID.
+func (s *statusServer) extractNodeID(ps httprouter.Params) (proto.NodeID, bool, error) {
+	nodeIDParam := ps.ByName("node_id")
+
+	// No parameter provided or set to local.
+	if len(nodeIDParam) == 0 || localRE.MatchString(nodeIDParam) {
+		return s.gossip.GetNodeID(), true, nil
+	}
+
+	id, err := strconv.ParseInt(nodeIDParam, 10, 64)
+	if err != nil {
+		return proto.NodeID(0), false, err
+	}
+	nodeID := proto.NodeID(id)
+	return nodeID, nodeID == s.gossip.GetNodeID(), nil
+}
+
+// proxy request performs a GET request to another node's status server.
+func (s *statusServer) proxyRequset(nodeID proto.NodeID, url string, w http.ResponseWriter, r *http.Request) {
```

Comment:
I feel like this should return an error and the response and that the caller should deal with responding to its client. I could be persuaded otherwise.